### PR TITLE
Update various PMIx URLs

### PR DESCRIPTION
**Pull request by jsquyres**
Signed-off-by: Jeff Squyres <jsquyres@cisco.com>

@rhc54 You should probably make sure I got all of these right...

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -28,12 +28,17 @@ When submitting questions and problems, be sure to include as much
 extra information as possible.  This web page details all the
 information that we request in order to provide assistance:
 
-     http://pmix.github.io/pmix/community/help/
+     https://pmix.org/support/
 
 The best way to report bugs, send comments, or ask questions is to
-sign up on the PMIx mailing list, which is hosted by GoogleGroups:
+post them on the OpenPMIX GitHub issue tracker:
```
Comment (rhc54) [inline README:34]: Putting on my Jeff Squyres "nit-picker" hat: it should be "OpenPMIx" - the "x" is not capitalized.

Comment (jsquyres) [reply README:34]: Fair point! ;-)

Comment (jsquyres) [reply README]: Fixed.

**Review by rhc54 — APPROVED** [review]
Thanks - deeply appreciated!

**Comment by jsquyres** [comment]
There's one CI failure, but obviously not related to this PR:

```
Command: cd /home/pmixer/scratch/pmix-v2.2/test ; timeout --preserve-status -k 35 30 ./pmix_test -n 5 --test-resolve-peers --ns-dist "1:2:2" -e /home/pmixer/scratch/pmix-v2.2/test/pmix_client
Status : 5 ***FAILED***
-----> : Test output
[f50459351975:47437] PMIX ERROR: NOT-FOUND in file preg_native.c at line 548
[f50459351975:47438] PMIX ERROR: NOT-FOUND in file preg_native.c at line 548
ERROR [test_resolve_peers.c:27:resolve_nspace]: smoky_nspace-0:0: Resolve peers test failed: rc = -47
ERROR [test_resolve_peers.c:27:resolve_nspace]: smoky_nspace-1:0: Resolve peers test failed: rc = -47
ERROR [test_resolve_peers.c:71:test_resolve_peers]: smoky_nspace-0:0: Resolve peers failed for the own namespace

ERROR [test_resolve_peers.c:71:test_resolve_peers]: smoky_nspace-1:0: Resolve peers failed for the own namespace

ERROR [pmix_client.c:165:main]: smoky_nspace-0:0 Resolve peers test failed: -1
ERROR [pmix_client.c:165:main]: smoky_nspace-1:0 Resolve peers test failed: -1
[f50459351975:47441] PMIX ERROR: NOT-FOUND in file preg_native.c at line 548
[f50459351975:47439] PMIX ERROR: NOT-FOUND in file preg_native.c at line 548
ERROR [test_resolve_peers.c:27:resolve_nspace]: smoky_nspace-2:1: Resolve peers test failed: rc = -47
ERROR [test_resolve_peers.c:27:resolve_nspace]: smoky_nspace-1:1: Resolve peers test failed: rc = -47
ERROR [test_resolve_peers.c:71:test_resolve_peers]: smoky_nspace-2:1: Resolve peers failed for the own namespace

ERROR [test_resolve_peers.c:71:test_resolve_peers]: smoky_nspace-1:1: Resolve peers failed for the own namespace

ERROR [pmix_client.c:165:main]: smoky_nspace-2:1 Resolve peers test failed: -1
ERROR [pmix_client.c:165:main]: smoky_nspace-1:1 Resolve peers test failed: -1
[f50459351975:47440] PMIX ERROR: NOT-FOUND in file preg_native.c at line 548
ERROR [test_resolve_peers.c:27:resolve_nspace]: smoky_nspace-2:0: Resolve peers test failed: rc = -47
ERROR [test_resolve_peers.c:71:test_resolve_peers]: smoky_nspace-2:0: Resolve peers failed for the own namespace

ERROR [pmix_client.c:165:main]: smoky_nspace-2:0 Resolve peers test failed: -1
```

Is that being worked on elsewhere?

**Comment by rhc54** [comment]
First I've seen it - not sure where that came from. All the PRs have been passing. I'll have to take a look, but no reason to delay this for it.

**Pull request merged**